### PR TITLE
ANN: don't suggest adding `unsafe` to main function in build scripts

### PR DESCRIPTION
**Pull request by Undin**
Also:
- introduce a proper API to check if the function is `main` one. Use it in `AddUnsafeFix` to support all cases and not to depend on `org.rust.cargo.runconfig` package
- don't show run line marker for `main` function in file with `no_main` attribute

Improvement of #8571

changelog: Don't show run line marker for `main` function in file with [`no_main`](https://doc.rust-lang.org/reference/crates-and-source-files.html#the-no_main-attribute) attribute

### CODE REVIEW

**Comment by Undin** [comment]
CC @Stzx

File: src/main/kotlin/org/rust/lang/core/psi/ext/RsFunction.kt [hunk 1]
```diff
@@ -31,6 +31,13 @@ val RsFunction.isMethod: Boolean get() = hasSelfParameters && owner.isImplOrTrai
 val RsFunction.isTest: Boolean
     get() = queryAttributes.isTest
 
+val RsFunction.isMain: Boolean get() {
```
Comment (Stzx) [inline src/main/kotlin/org/rust/lang/core/psi/ext/RsFunction.kt:34]: Maybe also provide a new function to detect the `#![no_main]` attribute?

Comment (Undin) [reply src/main/kotlin/org/rust/lang/core/psi/ext/RsFunction.kt:34]: It's a good idea. Thanks!

**Review by mchernyavsky — APPROVED** [review]
bors r+

**Pull request merged**